File: backend/routes/report.py
from fastapi import APIRouter
from fastapi.responses import FileResponse
from docx_generator import generate_case_report

router = APIRouter()


@router.post("/generate-report")
def generate_report(payload: dict):
    evidence = payload.get("evidence", "")
    analysis = payload.get("analysis", {})
    law = payload.get("law", {})
    cross = payload.get("cross", "")

    filename = generate_case_report(evidence, analysis, law, cross)
    return FileResponse(
        filename,
        media_type="application/vnd.openxmlformats-officedocument.wordprocessingml.document",
        filename=filename,
    )

File: backend/routes/report.js
const express = require("express");
const generateCaseReport = require("../docxGenerator");
const router = express.Router();

router.post("/generate-report", async (req, res) => {
  const { evidence, analysis, recommendations, cross } = req.body;

  const filename = await generateCaseReport(
    evidence,
    analysis,
    recommendations,
    cross
  );
  res.download(filename);
});

module.exports = router;
